### PR TITLE
[REF] Extract mapping converter function, kinda brutally

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Crude extraction of code to convert QF format to savable params

Before
----------------------------------------
Less re-usable

After
----------------------------------------
More re-usable

Technical Details
----------------------------------------
This is pretty brutal - I've extracted it & there is a test but it's just casting to an array at this stage - once we cleanup the inputs & outputs this might work with our new bulkSave....

Comments
----------------------------------------
@colemanw

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Core/BAO/Mapping.php [hunk 1]
```diff
@@ -990,6 +990,80 @@ public static function addComponentFields(&$fields, $mappingType, $exportMode) {
...
+   * @param array $params
+   * @param int $mappingId
+   * @param array $v
+   * @param string $k
+   * @param string $key
+   * @param int $colCnt
+   *
+   * @return array
+   */
+  protected static function getMappingParams($params, $mappingId, $v, $k, $key, int $colCnt): array {
```
Comment (colemanw) [inline CRM/Core/BAO/Mapping.php]: This is the first time I saw this syntax with a colon and return type. Neat. But it only works in php 7.

    CRM_Upgrade_Form::MINIMUM_PHP_VERSION = 5.6

Comment (eileenmcnaughton) [reply CRM/Core/BAO/Mapping.php]: hmm we HAVE dropped php 5.6 support though - maybe we should update that

File: CRM/Core/BAO/Mapping.php [hunk 2]
```diff
@@ -1222,67 +1296,15 @@ public static function saveMappingFields(&$params, $mappingId) {
     foreach ($params['mapper'] as $key => $value) {
       $colCnt = 0;
       foreach ($value as $k => $v) {
+        $locationTypeid = NULL;
+        $phoneTypeid = NULL;
```
Comment (colemanw) [inline CRM/Core/BAO/Mapping.php]: These 2 vars appear unused.

**Comment by colemanw** [comment]
@eileenmcnaughton I agree this doesn't have to be pretty as I don't expect this function to be around too long. Once the new exportUi extension is merged into core, the old quickform format won't be used at all & we can discard this.

File: CRM/Core/BAO/Mapping.php (hunk 2 above)
Comment (eileenmcnaughton) [reply CRM/Core/BAO/Mapping.php]: but they protect from enotices

Comment (colemanw) [reply CRM/Core/BAO/Mapping.php]: I read the code again and still don't see how they are doing anything in this spot. I see how variables with the same name are used in the new function, but not here.

Comment (eileenmcnaughton) [reply CRM/Core/BAO/Mapping.php]: ah maybe it was a refactor ago - push a change if you want to merge it now - I'm off to bed. The tests were failing on the enotice originally so it passes tests then all good

**Pull request merged**